### PR TITLE
#35 db manager getting tablenames

### PR DESCRIPTION
**Pull request by kortov**
Connect to #35 
- added corresponding method
- added schema field to DBmanager
- added tests

### CODE REVIEW

File: src/main/java/juja/sqlcmd/DatabaseManager.java [hunk 1]
```diff
@@ -30,4 +43,21 @@ public boolean connect(String database, String user, String password) {
...
+    }
+
+    public String[] getTableNames() throws SQLException {
+        try (ResultSet rs = connection.getMetaData().getTables(null, schemaPattern, "%", new String[]{"TABLE"})) {
+            List<String> tablesList = new LinkedList<>();
+            while (rs.next()) {
+                tablesList.add(rs.getString("table_name"));
+            }
+            return tablesList.toArray(new String[0]);
+        }
```
Comment (AVM82) [inline src/main/java/juja/sqlcmd/DatabaseManager.java]: I'm not sure that such an implementation getTableNames() corresponds to the requirement of the task.

Comment (kortov) [reply src/main/java/juja/sqlcmd/DatabaseManager.java]: Yes, I know it’s cheating, I wrote about it in Slack, but anyway, my hands couldn’t write rough code with arrays..))

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 2]
```diff
@@ -33,34 +63,62 @@ public void cleanUpStreams() {
     }
 
     @Test
-    public void testConnectWithAllValidParameters() {
-        assertTrue(databaseManager.connect("sqlcmd", "sqlcmd", "sqlcmd"));
+    public void ConnectWithAllValidParameters() {
```
Comment (adaleorion) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: I think name should be `connectWithAllValidParameters`
First letter is lowercase

File: src/main/java/juja/sqlcmd/DatabaseManager.java [hunk 3]
```diff
@@ -30,4 +43,21 @@ public boolean connect(String database, String user, String password) {
             return false;
         }
     }
+
+    public static void main(String[] args) throws SQLException {
```
Comment (adaleorion) [inline src/main/java/juja/sqlcmd/DatabaseManager.java]: What for do you need this method main ?

**Comment by adaleorion** [comment]
Please use #number-task  for connection this pull request to task

File: src/main/java/juja/sqlcmd/DatabaseManager.java (hunk 1 above)
Comment (adaleorion) [reply src/main/java/juja/sqlcmd/DatabaseManager.java]: I agree with @AVM82 
Please use simple implementation

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 4]
```diff
@@ -1,29 +1,59 @@
...
+    private static Connection connection;
     private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
     private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
-    private DatabaseManager databaseManager;
     private PrintStream originalOut;
     private PrintStream originalErr;
 
+    @BeforeClass
+    public static void setTestingEnvironment() throws SQLException {
+        databaseManager = new DatabaseManager();
```
Comment (adaleorion) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: What for do you need creation databaseManager right here ?

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 5]
```diff
@@ -33,34 +63,62 @@ public void cleanUpStreams() {
...
+    public void getTableNamesWith2tables() throws SQLException {
+        executeSqlQuery("CREATE TABLE " + SCHEMA_PATTERN + ".table1()");
+        executeSqlQuery("CREATE TABLE " + SCHEMA_PATTERN + ".table2()");
+        String[] expectedArray = {"table1", "table2"};
+        String[] actualArray = databaseManager.getTableNames();
+        assertArrayEquals(expectedArray, actualArray);
+    }
+
+    @Test
+    public void getTableNamesReturnsArrayWith2Tables() throws SQLException {
```
Comment (adaleorion) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: This is strange name for test.  
This name doesn't respond body method

File: src/main/java/juja/sqlcmd/DatabaseManager.java [hunk 6]
```diff
@@ -2,13 +2,26 @@
...
+import java.util.List;
 
 public class DatabaseManager {
     private final String jdbcDriverClass;
     private final String jdbcUrl;
+    private String schemaPattern = "public";
     private Connection connection;
 
+    public String getSchemaPattern() {
+        return schemaPattern;
```
Comment (adaleorion) [inline src/main/java/juja/sqlcmd/DatabaseManager.java]: I think name `schemaName` is better.
Maybe schema name should be paramert for gettables(),
What do you think ?

File: src/main/java/juja/sqlcmd/DatabaseManager.java (hunk 1 above)
Comment (kortov) [reply src/main/java/juja/sqlcmd/DatabaseManager.java]: @AVM82 @danilkuznetsov sorry I really don't see any reasons for arrays. Metadata is a default implementation for retrieving tables from databases. It's database independent. But this implementation does not imply neither Count(*) nor ResultSet.last methods for creating new Array[size].

Comment (Shtramak) [reply src/main/java/juja/sqlcmd/DatabaseManager.java]: Using next trick you can get a size of your ResultSet: 
`int size = 0;
if (rs.last()) {
    size = rs.getRow();
    rs.beforeFirst();
}`

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 7]
```diff
@@ -1,29 +1,59 @@
...
+import static org.junit.Assert.assertArrayEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
 public class DatabaseManagerTest {
+    private static final String TEST_DB_NAME = "test";
+    private static final String DB_USER_NAME = "sqlcmd";
+    private static final String DB_USER_PASSWORD = "sqlcmd";
+    private static final String JDBC_URL = "jdbc:postgresql://localhost:5432/";
+    private static DatabaseManager databaseManager;
```
Comment (Shtramak) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: There's no need databaseManager to be static, you're using it only in non static context.

File: src/test/java/juja/sqlcmd/DatabaseManagerTest.java [hunk 8]
```diff
@@ -1,29 +1,59 @@
...
+    private static Connection connection;
     private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
     private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
-    private DatabaseManager databaseManager;
     private PrintStream originalOut;
```
Comment (Shtramak) [inline src/test/java/juja/sqlcmd/DatabaseManagerTest.java]: static and non static blocks should be separate by blank line (http://bit.ly/2j2qPE1)

**Review by Shtramak — APPROVED** [review]
Looks good to me!

**Review by adaleorion — APPROVED** [review]
looks nice :+1:

**Review by AVM82 — APPROVED** [review]

**Pull request merged**